src/System.Runtime.Extensions/src/System/IO/PathHelper.Windows.cs

```diff
@@ -223,15 +191,15 @@ private StringBuilder GetFullPathName(string path)
                 Debug.Assert(!PathInternal.IsExtended(path));
             }
 
-            StringBuilder outputBuffer = this.GetOutputBuffer(capacity);
+            StringBuffer outputBuffer = new StringBuffer((ulong)capacity);
```

Comment:
I started with the pool on the StringBuffer itself and slowly gravitated towards putting it internal. Made usage way easier. In addition spreading caching across all interop seems much more scalable. Keeping a single cache for interop would seem to scale to different mixes of interop better (string/byte/UTF8).